src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java

```diff
@@ -44,11 +47,29 @@ public class MT940FormatValidator extends AbstractConnector {
     @Override
     public void connect(MessageContext messageContext) throws ConnectException {
         log.debug("Executing MT940FormatValidator to validate the MT940 format");
-        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) messageContext)
-                .getAxis2MessageContext();
-        Optional<String> payload = ConnectorUtils.buildMessagePayloadFromMessageContext(axis2MessageContext);
-        if (payload.isPresent()) {
-            String[] lines = payload.get().split(ConnectorConstants.LINE_BREAK);
+
+        String payload = null;
+        try {
+            JSONObject jsonPayload = XML.toJSONObject(messageContext.getEnvelope().getBody().toString())
+                    .getJSONObject("soapenv:Body");
+            if (jsonPayload.has("text")) {
+                payload = jsonPayload.getJSONObject("text").getString("content");
+            } else {
+                for (Iterator it = jsonPayload.keys(); it.hasNext(); ) {
+                    String key = (String) it.next();
+                    if (key.startsWith("axis2ns")) {
+                        payload = jsonPayload.getJSONObject(key).getString("content");
+                    }
```

Comment:
Can we break the loop here? also its clean if we could use streams for this
```suggestion
                    if (key.startsWith("axis2ns")) {
                        payload = jsonPayload.getJSONObject(key).getString("content");
                        break;
                    }
```

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/7